fields/main.go

```diff
@@ -130,299 +221,185 @@ func main() {
-	if err != nil {
-		return nil, err
-	}
+	empty := &FieldInfo{}
 
-	if len(respBody.Data) != 1 {
-		return nil, &NotFoundError{}
-	}
-
-	d := respBody.Data[0]
-	return &d, nil
-}
-
-func (c *Client) delete%[1]s(ctx context.Context, site, id string) error {
-	err := c.do(ctx, "DELETE", fmt.Sprintf("s/%%s/rest/%[2]s/%%s", site, id), struct{}{}, nil)
-	if err != nil {
-		return err
-	}
-	return nil
-}
-
-func (c *Client) create%[1]s(ctx context.Context, site string, d *%[1]s) (*%[1]s, error) {
-	var respBody struct {
-		Meta meta `+"`"+`json:"meta"`+"`"+`
-		Data []%[1]s `+"`"+`json:"data"`+"`"+`
-	}
-
-	err := c.do(ctx, "POST", fmt.Sprintf("s/%%s/rest/%[2]s", site), d, &respBody)
-	if err != nil {
-		return nil, err
-	}
-
-	if len(respBody.Data) != 1 {
-		return nil, &NotFoundError{}
-	}
-
-	new := respBody.Data[0]
-
-	return &new, nil
-}
-
-func (c *Client) update%[1]s(ctx context.Context, site string, d *%[1]s) (*%[1]s, error) {
-	var respBody struct {
-		Meta meta `+"`"+`json:"meta"`+"`"+`
-		Data []%[1]s `+"`"+`json:"data"`+"`"+`
-	}
-
-	err := c.do(ctx, "PUT", fmt.Sprintf("s/%%s/rest/%[2]s/%%s", site, d.ID), d, &respBody)
-	if err != nil {
-		return nil, err
-	}
-
-	if len(respBody.Data) != 1 {
-		return nil, &NotFoundError{}
-	}
-
-	new := respBody.Data[0]
-
-	return &new, nil
-}
-`, structName, urlPath)
-
-	return code, nil
-}
-
-func normalizeValidation(re string) string {
-	re = strings.ReplaceAll(re, "\\d", "[0-9]")
-	re = strings.ReplaceAll(re, "[-+]?", "")
-	re = strings.ReplaceAll(re, "[+-]?", "")
-	re = strings.ReplaceAll(re, "[-]?", "")
-	re = strings.ReplaceAll(re, "\\.", ".")
-	re = strings.ReplaceAll(re, "[.]?", ".")
-
-	quants := regexp.MustCompile(`\{\d*,?\d*\}|\*|\+|\?`)
-	re = quants.ReplaceAllString(re, "")
-
-	control := regexp.MustCompile(`[\(\[\]\)\|\-\$\^]`)
-	re = control.ReplaceAllString(re, "")
-
-	re = strings.TrimPrefix(re, "^")
-	re = strings.TrimSuffix(re, "$")
-
-	return re
-}
-
-func typeFromValidation(validation interface{}) (ty string, comment string, omitempty bool, err error) {
 	switch validation := validation.(type) {
 	case []interface{}:
 		if len(validation) == 0 {
-			return "[]string", "", false, nil
+			return NewFieldInfo(fieldName, name, "string", "", false, true), nil
 		}
 		if len(validation) > 1 {
-			return "", "", false, fmt.Errorf("unknown validation %#v", validation)
+			return empty, fmt.Errorf("unknown validation %#v", validation)
 		}
-		elementType, elementComment, _, err := typeFromValidation(validation[0])
+
+		fieldInfo, err := r.fieldInfoFromValidation(name, validation[0])
 		if err != nil {
-			return "", "", false, err
+			return empty, err
 		}
-		return fmt.Sprintf("[]%s", elementType), elementComment, true, nil
+
+		fieldInfo.OmitEmpty = true
+		fieldInfo.IsArray = true
+
+		return fieldInfo, nil
 	case map[string]interface{}:
-		fieldNames := []string{}
-		fieldCodes := map[string]string{}
+		typeName := r.StructName + "_" + fieldName
```

Comment:
I'd prefer to not use an underscore here, but if it results in conflicts its a reasonable compromise.

Reply:
I agree with this. I added the underscore to ensure there were no type naming conflicts, but I don't know if it's necessary. If there's a future conflict that arises, we could instead create a special rule for just that type.

I'll remove this and do some more testing. Thanks!